seq2science/rules/configuration_workflows.smk

```diff
@@ -67,6 +67,18 @@ for conf_dict in ["aligner", "quantifier", "diffexp"]:
             config[k] = v
         config[conf_dict] = dict_key
 
+
+# ...for rna-seq
+if get_workflow() == "rna_seq":
+    # delete the old strandedness report if samples.tsv was updated
+    strandedness_report = f"{config['qc_dir']}/strandedness/inferred_strandedness.tsv"
+    if os.path.exists(strandedness_report) and not config['ignore_strandedness']:
+        strandedness = pd.read_csv(strandedness_report, sep='\t', dtype='str', index_col=0)
+        col = samples.replicate if "replicate" in samples else samples.index
+        if len(strandedness.index) != len(set(col)) or not all(s in set(col) for s in strandedness.index):
+            os.unlink(strandedness_report)
+
```

Comment:
This is a bit iffy.. It breaks the flow of the rest of the snakemake code, by doing this with python. Currently it just sucks a bit the way snakemake resolves what needs to be run, so I am fine with it. 

Something for a blue monday in the far future: we could consider doing this (and other file dependencies) more properly. Not entirely sure what would be a good way to do it, but for instance checksumming the config and samples file, and using those checksums as inputs for rules or even some config options. However that is going to make the rules even more bulky, will introduce weird bugs, and is a lot of work.